protocols/imap/src/main/java/org/apache/james/imap/main/PathConverter.java

```diff
@@ -71,13 +71,28 @@ private boolean isADelegatedMailboxName(List<String> mailboxNameParts) {
 
     private MailboxPath buildDelegatedMailboxPath(List<String> mailboxNameParts) {
         return new MailboxPath(MailboxConstants.USER_NAMESPACE,
-            mailboxNameParts.get(USER_PART),
+            addDomainPartToAmbigusUserName(mailboxNameParts.get(USER_PART)),
             sanitizeMailboxName(
                 Joiner.on(pathDelimiter)
                     .skipNulls()
                     .join(Iterables.skip(mailboxNameParts, 2))));
     }
 
+    private String addDomainPartToAmbigusUserName(String otherUserName) {
```

Comment:
you basically want a User object to avoid such string concatenation

Reply:
DONE. See https://github.com/linagora/james-project/pull/1100